shared/missing.h

```diff
@@ -20,6 +20,7 @@
 #endif
 
 #ifndef __NR_finit_module
+# warning __NR_finit_module missing - kmod might not work correctly
```

Comment:
`missing.h: warn on missing __NR_finit_module`

> Another, perhaps better option, is to outright fail the build, since 
> if this triggers we have semi-broken kmod. Namely: rmmod will work, yet
> insmod would fail.

did we break the fallback to init_module() when finit_module() is missing?  By now I think we can start warning since it has been a decade since finit_module() was introduced, but I don't like outright breaking the build if we don't have a good reason to.